### DISCUSSION TITLE
Creating an organization

### DISCUSSION THREAD

**Discussion by iboB** [cpm-cmake/CPM.cmake#185] [General]
@TheLartians, have you thought about creating a GitHub organization for CPM?

The benefits are that this and related repos will be in the same namespace of sorts. Plus as this project grows (as I hope it will), it will be easier to add and manage collaborators without involving your other projects.

Ideas for additional repos include: helper GitHub actions (say cpm-cache as a separate action), a separate list of snippets for popular projects as discussed in the reddit thread.

**Comment by TheLartians** [comment]
I think it's a good idea, and probably makes sense at this point. One thing I would like to be certain of first is that urls to former releases still work (as used in `get_cpm.cmake`) and we won't be affecting everybody's workflow through the transfer.

**Reply by iboB** [reply]
Former releases should work through TheLartians/CPM.cmake 
This repo could still exist, but it could be archived with links in the readme to the organization main repo

**Reply by TheLartians** [reply]
Great to know, thanks! I was also planning to do some tests with a dummy repo later, to validate the behaviour. If all goes well, I'd like to transfer the original repo, as all issues, the wiki and discussions would be preserved.

**Reply by TheLartians** [reply]
Transferring releases indeed seem to work fine! 🎉 
I'll probably begin moving the CPM.cmake repo in the evening.

**Comment by iboB** [comment]
An organization would probably be a good place to host "CPM-ified" community repos of popular projects that don't have CMake builds, or do, but they're not suitable for CPM for some reason. 

Boost comes to mind. Also Qt, POCO, and probably more.

I'm now forking [Orphis/boost-cmake](https://github.com/Orphis/boost-cmake) to have the latest boost support, but an official boost CPM-friendly repo will probably be quite useful (this problem was also recently mentioned in #190 )

**Comment by TheLartians** [comment]
On that note, what do you think about the org name [_CPM-cmake_](https://github.com/orgs/CPM-cmake)? As GitHub doesn't allow the `.` character and _CPM_ is apparently already taken it seems like the closest to the original project.

The full repo path then should be `https://github.com/CPM-cmake/CPM.cmake`.

**Reply by iboB** [reply]
To be honest. I don't like how CPM-cmake looks from a pure aesthetic point of view.

A common pattern to name projects and organizations related to those projects for some languages is `<language>-<project>`, This pattern would lead to "cmake-CPM"... but I don't like this too

I think the problem for me is the mix of caps and lowercase. As a URL it doesn't matter which is chosen as GitHub URLs are case insensitive, but the page displays the original org name. So, I would *personally* prefer  "cpm-cmake" or "cmake-cpm", but ultimately it's your call and the burden of bikeshedding the name falls to you 😄 

PS: Just some random ideas from me in no particular order (I don't particularly like any of them)

CPM-CMake
CMake-CPM
CMakePackageManager
CPMcmake
C-P-M
CPM-org
theCPM
CPM-dev
cpm-dev

**Reply by iboB** [reply]
Actually CPMcmake is beginning to grow on me 😄

**Reply by TheLartians** [reply]
Thanks a lot for the feedback and suggestions!

Now I see that the mixing of uppercase and lowercase letters was bothering me too. Without renaming the main repo itself, I personally like the look of `https://github.com/cpm-cmake/CPM.cmake` most, so let's go with that! 😄

**Reply by TheLartians** [reply]
> Actually CPMcmake is beginning to grow on me 😄

Missed your post by a few seconds, I still stand by liking `cpm-cmake` most, but I'll give it some time to see if the other options do grow on me! 😅

In the meantime I'll be doing some release/transfer experiments.

**Comment by TheLartians** [comment]
Transfer complete! 🎉 

@iboB as you brought it up and are apparently working on CPM.cmake friendly projects, would you be interested in joining the organisation and helping maintain the project?

**Reply by iboB** [reply]
Sure, yes, thanks. I may not have all the time in the world these days, but I'll try to help

**Comment by ClausKlein** [comment]
Good Idea at all!

What is about cmake modules like your [PackageProject.cmake](https://github.com/TheLartians/PackageProject.cmake)?

Should it pushed to `cmake.org` as official cmake module?

Any plan?

**Reply by TheLartians** [reply]
Becoming an official CMake module would be great for exposure (if they are even interested), however there are also some advantages to being an independent project. For example:

- Faster iteration speed, independent of CMake releases
- Can add opinionated functionality e.g. CPM.cmake caches based on git tags, something CMake probably won't ever support as git tags are subject to change
- Add support for users stuck with old CMake versions

But I am definitely planning on moving more repos like PackageProject.cmake or [CPMLicences.cmake](https://github.com/cpm-cmake/CPMLicenses.cmake) to the cpm-cmake organisation.
